src/test/run-make-fulldeps/link-cfg/Makefile

```diff
@@ -2,7 +2,10 @@
 
 all: $(call DYLIB,return1) $(call DYLIB,return2) $(call NATIVE_STATICLIB,return3)
 	ls $(TMPDIR)
-	$(RUSTC) --print cfg --target x86_64-unknown-linux-musl | $(CGREP) crt-static
+
+    ifneq ($(IS_MUSL_HOST),1)
+	    $(RUSTC) --print cfg --target x86_64-unknown-linux-musl | $(CGREP) crt-static
+    endif
```

Comment:
How come this tests needed to change? I think the target probably doesn't want to change in this PR ideally as that may make it susceptible to having an accidentally breaking change.

Reply:
I feel bad for it but we have `RUSTFLAGS="-C target-feature=-crt-static"` defined all the time so our static musl toolchain becomes dynamic toolchain.
Again using something like `dynmusl` target for the host would keep `musl` static for whole test suite.

I'll try replacing `$(RUSTC)` with `$(BARE_RUSTC)` so `RUSTFLAGS` are ignored.